### DISCUSSION TITLE
Some insights and solutions: Data visualization and package dependencies

### DISCUSSION THREAD

**Discussion by emmyxth** [CS342/.github#32] [Technical Questions] [answered]
### In what area do you have a technical challenge?

Swift Package Manager

### Description

For some of our projects, we want to visualize our users' data. How do we do that? 

Initially, our team considered using ResearchKit and CareKit. Both seem versatile and well-documented by Apple, as seen from these tutorials. After attempting to install them using these[ official instructions ](https://github.com/carekit-apple/CareKit/blob/master/README.md)however, we ran into a "Package not found" error.

This is because we need to install it into the app itself– we need to create dependencies to our main target.

We can do so by:
- Navigating to {TemplateApplication}Modules/Package.swift
- Adding dependencies to the `dependencies` list in `Package()` in the format `.package(url: {url here}, .upToNextMinor(from: {version here}))`
- If applicable, going to the module in which you are going to be using the packages in (such as '{TemplateApplication}Trends') and adding dependencies in the form `.product(name: {package name}, package: {package name})`. 

Something else we realized is that we could circumvent ALL of this by using [Swift charts](https://developer.apple.com/documentation/charts): much more documentation, lightweight, and just works better for our purposes. But again, it depends on your project's goals!

### Reproduction

When you try installing package dependencies but it appears to not be universal.

### Expected behavior

It should be universal

### Code of Conduct

- [X] I agree to follow this course's Code of Conduct and Contributing Guidelines

**Comment by PSchmiedmayer** [comment, marked as answer]
@emmyxth Thank you for sharing this with the class! 🚀
